### ISSUE TITLE
Test run fails under eclipse

### ISSUE THREAD

**Issue by Nuclearfarts** [QuiltMC/quilt-loader#4]
This is a regression from latest fabric-loader. Running via gradle works fine, as well.

Crash log: (Note: `TestMod` does implement `PreLaunchEntrypoint`, but it's loaded by `KnotClassLoader`, and the `PreLaunchEntrypoint` referenced in the crash report is loaded by the system classloader.)

```
Exception in thread "main" java.lang.RuntimeException: Could not execute entrypoint stage 'preLaunch' due to errors, provided by 'test'!
	at org.quiltmc.loader.impl.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:50)
	at org.quiltmc.loader.impl.entrypoint.minecraft.hooks.EntrypointUtils.invoke(EntrypointUtils.java:33)
	at org.quiltmc.loader.impl.launch.knot.Knot.init(Knot.java:138)
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28)
	at org.quiltmc.devlaunchinjector.Main.main(Main.java:86)
Caused by: net.fabricmc.loader.api.EntrypointException: Exception while loading entries for entrypoint 'preLaunch' provided by 'test'
	at org.quiltmc.loader.impl.EntrypointStorage.lambda$1(EntrypointStorage.java:186)
	at org.quiltmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:37)
	at org.quiltmc.loader.impl.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:47)
	... 4 more
Caused by: net.fabricmc.loader.api.LanguageAdapterException: Class net.fabricmc.test.TestMod cannot be cast to net.fabricmc.loader.api.entrypoint.PreLaunchEntrypoint!
	at org.quiltmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:58)
	at org.quiltmc.loader.impl.EntrypointStorage$NewEntry.create(EntrypointStorage.java:112)
	at org.quiltmc.loader.impl.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:99)
	at org.quiltmc.loader.impl.EntrypointStorage.lambda$1(EntrypointStorage.java:184)
	... 6 more```

**Comment by AlexIIL** [comment]
Just to check, what version or commit of quilt-loader are you using? (Also, does #3 fix this? It did for me).

**Comment by Nuclearfarts** [comment]
Yep, closed by #3
